src/main/resources/hudson/plugins/timestamper/annotator/TimestampAnnotatorFactory2/script.js

```diff
@@ -48,16 +48,21 @@ function init() {
     };
 
     // Set the mode from a cookie or initialize it (also handle migrating old cookie values to new ones).
-    var mode = getCookie() || 'system';
+    var mode = getCookie();
+    if (!mode) {
+        // Initialize the cookie, defaulting to clock time in the browser's timezone.
+        mode = 'system';
+
+        setCookie(mode);
+        setCookie('true', 'local');
+    }
+
     if (mode == 'local') {
         options[mode].checked = true;
         mode = 'system';
     }
     modes[mode].checked = true;
 
-    // Renew the cookie.
-    setCookie(mode);
```

Comment:
Not sure this should have been removed? It was there to auto-renew the cookie, so that it only expires if the page has not been visited in 2 years as in https://googleblog.blogspot.com.au/2007/07/cookies-expiring-sooner-to-improve.html

Reply:
Updated, thanks.